packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -293,6 +294,21 @@ export default class AssetGraphBuilder extends EventEmitter {
       ...input,
       optionsRef: this.optionsRef,
     });
+
+    // Add invalidations to the request if a node already exists in the graph.
+    // These are used to compute the cache key for assets during transformation.
+    if (this.requestGraph.hasNode(request.id)) {
+      request.input.invalidations = this.requestGraph
```

Comment:
Unsure about mutating the request input here. I imagine this would also get stored in the graph as part of the request. @padmaia do you think we need a way to send arguments to the request function without storing them?

Reply:
Yeah, this feels a bit wrong. And yes, it would end up getting stored in the graph, though we are storing inputs in the graph without much benefit right now. I added that functionality when I was thinking that we might want to rerun all invalidated in requests in parallel, but what I've found is that it's much simpler and often more performant to just start at the top of the request graph again and skip requests that are still valid because requests further up the graph will end up deleting requests further down. So maybe we should think about no longer storing inputs, but that doesn't really solve the problem of needing to pass the invalidations in to the Transformation.

One idea I've had is that we could pass previous results into requests, which would give us the option of iterating on the previous result instead of completely rebuilding it. Was thinking this type of behavior would be useful for an `AssetGraphRequest` once we get there. I know we are trying to keep the graph thin, but since results are being cached we could maybe just store the cache key(s). Might take some refactoring in Transformation, but I think this type of approach would make the code less confusing.

Reply:
The easiest option would be to pass the request graph into request handlers. Then the asset request itself could query the graph for invalidations.

Reply:
Updated to do that: ab81fe8.

Reply:
Requests would be able to do some unsafe things with full access to the graph. What if you added the `getInvalidations` function to the API to passed into requests?

Reply:
Great idea! Done. 😄